File: 122. Best Time to Buy and Sell Stock II.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.empty())
            return 0;
        int sumProfit = 0;
        int profit = 0;
        int lastPrice = prices[0];
        for (auto p : prices) {
            if (p > lastPrice) 
                profit += p - lastPrice;
            if (p < lastPrice) {
                sumProfit += profit;
                profit = 0;
            }
            lastPrice = p;
        }
        sumProfit += profit;
        return sumProfit;
    }
};

File: 122. Best Time to Buy and Sell Stock II.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len( prices) <= 1:
            return 0
        profit = 0
        last = prices[0]
        for price in prices[1:]:
            profit += max(0, price - last)
            last = price
        return profit
            
        
